### PR TITLE
Refactoring of Format data

### PR DESCRIPTION
**Pull request by CasperKjaerhus**
Enclose functions in static class of FCD_Formatter.

This class is responsible for taking the original FCD data from the API and converting it into a pandas dataframe of the form:

index | osm_id | coordinates

SIDE NOTE: 
I've also fixed the dockerfile to install requirements as well as gitignoring the python install files in the .egg-info folder

### CODE REVIEW

**Review by anesheim-uni — APPROVED** [review]
Sure! Here's an example of an approval message for a colleague's pull request:

Hi [Colleague's Name],

I just wanted to let you know that I had a chance to review your pull request, and I'm really impressed with the work you've done. Your code is well-written, easy to understand, and clearly demonstrates your expertise in this area. I especially appreciate the way you [mention a specific aspect of their work that you found particularly impressive].

Overall, I think your contributions have greatly improved our project, and I'm excited to see this work merged into the main codebase. Thanks so much for your hard work and dedication!

Best regards,
[Your Name]

File: pipeline/preprocessing/formatting/FCD_Formatter.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        wrapdata (Any): The json object from the AAU FCD Request
+
+    Returns:
+        DataFrame: the "meat" of the FCD Data
+    """
+    df = DataFrame.from_records(  # convert json file to dataframe
+        wrapdata["results"]["features"]
+    )
+
+    # unnest some of the nested values
```
Comment (frejaz) [inline pipeline/preprocessing/formatting/FCD_Formatter.py:60]: Funktionen unnest_df() gør det her i process_merged_geojson.py

**Review by frejaz — CHANGES_REQUESTED** [review]
lille funktion

File: dockerfile [hunk 2]
```diff
@@ -2,6 +2,7 @@ FROM python:3.11-slim
 
 RUN apt-get update
 
-RUN pip3 install requests
+COPY requirements.txt .
+RUN pip3 install -r requirements.txt
```
Comment (AlexanderHHansen) [inline dockerfile:6]: Kan du tilføje 
```
python -m pip install -e .
```

File: pipeline/preprocessing/formatting/FCD_Formatter.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            data
+        )  # Remove the outer layer of json object
+        data = _create_segment_to_coordinate_df(
+            data
+        )  # Convert from geojson format to osm_id -> trips
+
+        return data
+
+
+def _remove_fcd_request_wrapper(wrapdata: Any) -> DataFrame:
```
Comment (AlexanderHHansen) [inline pipeline/preprocessing/formatting/FCD_Formatter.py:47]: Private functions skal have double underscore. én underscore indikerer protected.

Comment (AlexanderHHansen) [reply pipeline/preprocessing/formatting/FCD_Formatter.py:47]: Samme for de resterende funktioner

File: pipeline/preprocessing/formatting/FCD_Formatter.py (hunk 1 above)
Comment (CasperKjaerhus) [reply pipeline/preprocessing/formatting/FCD_Formatter.py:61]: 👍

File: pipeline/preprocessing/formatting/FCD_Formatter.py (hunk 3 above)
Comment (CasperKjaerhus) [reply pipeline/preprocessing/formatting/FCD_Formatter.py:48]: Så kan vi ik test

File: dockerfile (hunk 2 above)
Comment (CasperKjaerhus) [reply dockerfile:6]: 👍

**Review by frejaz — APPROVED** [review]
noice

**Pull request merged**